lib/http1.c

```diff
@@ -529,8 +544,7 @@ static void reqread_on_timeout(h2o_timer_t *entry)
         conn->req.res.reason = "Request Timeout";
     }
 
-    conn->req.http1_is_persistent = 0;
-    close_connection(conn, 1);
+    close_idle_connection(&conn->super);
```

Comment:
That's a good point. I do assume that `conn->sock->input->size == 0` because `reqread_on_timeout` fired, which means there was nothing to read and we ended up timing out. I'll add the assert.

Reply:
Don't we see a timeout with a non-empty `conn->sock->input` if the client has sent a partial HTTP2 frame?

Reply:
> Don't we see a timeout with a non-empty conn->sock->input if the client has sent a partial HTTP2 frame?

Would that happen in http1? In a connection upgrade?

Reply:
My bad, please ignore my comment here.